frontend/model/contracts/group.js

```diff
@@ -1810,16 +1747,26 @@ sbp('chelonia/defineContract', {
           encryptionKeyId,
           ...actorID === memberID && {
             hooks: {
+              postpublish: () => {
+                sbp('chelonia/contract/retain', chatRoomId)
+              },
               onprocessed: () => {
                 sbp('state/vuex/commit', 'setCurrentChatRoomId', { groupId: contractID, chatRoomId })
               }
             }
           }
         }).catch(e => {
+          if (e.name === 'GIErrorUIRuntimeError' && e.cause?.name === 'GIChatroomAlreadyMemberError') {
+            if (actorID === memberID) {
+              // Increase reference count if we've already joined
+              sbp('chelonia/contract/retain', chatRoomId)
+            }
+            return
+          }
```

Comment:
In what sort of situation would we want to increase the retain count here again?

Reply:
When re-syncing contracts (e.g., from a new device), as the reference count would be 0 and the action would fail to send with `GIChatroomAlreadyMemberError`.

Reply:
The `gi.actions/chatroom/join` action, of course. And since it failed, that's why this is in the `.catch` block, to see why it failed.

Reply:
So, we are trying to join a chatroom that we are already a part of. If it we are already a part of it, why would the reference count be `0`?

Reply:
I see.... OK. This too, is super unintuitive. Although this code already has a comment, could you please expand on it to make it very clear about this specific situation, to mention that this is the situation where this a new device syncing from scratch to a channel we've already joined on another one?